include/service/entities/custom_object_service.js

```diff
@@ -955,7 +955,7 @@ module.exports = function CustomObjectServiceModule(pb) {
             options = {};
         }
 
-        if (!pb.validation.isIdStr(id, true)) {
+        if (!pb.validation.isIdStr(id.toString(), true)) {
```

Comment:
That is incorrect.  The ```ValidationService.isId``` function checks both Strings and ObjectID objects.  ```ValidationService.isIdStr``` adds the additional constraint that the value must also be a string as well as parseable into an ObjectID object.  If you look at the implementation of the ```isIdStr``` function you'll see that it validates the value as a string then calls the ```isId``` function.